tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -347,6 +350,74 @@ void SetHtmlResponseContent()
             return blockingAction;
         }
 
+        private string GetJsonResponse()
+        {
+            try
+            {
+                if (_blockedJsonTemplateCache != null)
+                {
+                    return _blockedJsonTemplateCache;
+                }
+
+                if (_settings.BlockedJsonTemplatePath != null && AppDomain.CurrentDomain.BaseDirectory != null)
+                {
+                    var fullPath =
+                        Path.IsPathRooted(_settings.BlockedJsonTemplatePath)
+                            ? _settings.BlockedJsonTemplatePath
+                            : Path.Combine(AppDomain.CurrentDomain.BaseDirectory, _settings.BlockedJsonTemplatePath);
+                    if (File.Exists(fullPath))
+                    {
+                        _blockedJsonTemplateCache = File.ReadAllText(fullPath);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error setting json blocking template, defaulting in built in template");
+            }
+
+            if (_blockedJsonTemplateCache == null)
+            {
+                _blockedJsonTemplateCache = SecurityConstants.BlockedJsonTemplate;
+            }
+
+            return _blockedJsonTemplateCache;
```

Comment:
This method obviously isn't thread safe, but I assume that's not a concern

Reply:
I believe it's not an issue as the method will be deterministic no matter how many threads enter it at the same time. It could result in the template file be read many times till the values finally cached, but this probably better than introducing synchronization.